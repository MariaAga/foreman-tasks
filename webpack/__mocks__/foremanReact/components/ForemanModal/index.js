import React from 'react';

const ForemanModal = () => <div>Modal</div>;
ForemanModal.Header = () => <div>Modal Header</div>;
ForemanModal.Footer = () => <div>Modal Footer</div>;
export default ForemanModal;
