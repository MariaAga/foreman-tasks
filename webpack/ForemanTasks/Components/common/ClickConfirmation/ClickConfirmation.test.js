import React from 'react';
import { testComponentSnapshotsWithFixtures, mount } from '@theforeman/test';
import { useForemanModal } from 'foremanReact/components/ForemanModal/ForemanModalHooks';
import { ClickConfirmation } from './';

const fixtures = {
  render: {
    title: 'some-title',
    confirmType: 'danger',
    body: 'some-body',
    confirmationMessage: 'some-message',
    id: 'some-id',
    confirmAction: 'some-confirm',
    onClick: jest.fn(),
  },
};

describe('ClickConfirmation', () => {
  testComponentSnapshotsWithFixtures(ClickConfirmation, fixtures);
  it('enable button on checkbox click', () => {
    const component = mount(<ClickConfirmation {...fixtures.render} />);
    const getButton = () => component.find('.confirm-button').at(0);
    expect(getButton().props().disabled).toBeTruthy();
    const checkbox = component.find('input').at(0);
    checkbox.simulate('change', { target: { checked: true } });
    expect(getButton().props().disabled).toBeFalsy();
  });

  it('click test', () => {
    const setModalClosed = jest.fn();
    useForemanModal.mockImplementation(id => ({
      setModalClosed: () => setModalClosed(id),
    }));
    const onClick = jest.fn();
    const props = { ...fixtures.render, onClick };
    const component = mount(<ClickConfirmation {...props} />);
    const getButton = () => component.find('.confirm-button').at(0);
    const checkbox = component.find('input').at(0);
    checkbox.simulate('change', { target: { checked: true } });
    getButton().simulate('click');
    expect(onClick).toBeCalled();
    expect(setModalClosed).toBeCalledWith({ id: fixtures.render.id });
  });
});
