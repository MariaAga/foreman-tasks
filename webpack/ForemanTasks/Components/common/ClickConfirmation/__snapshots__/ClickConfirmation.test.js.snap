// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickConfirmation render 1`] = `
<ForemanModal
  id="some-id"
>
  <Component>
    <span
      className="glyphicon glyphicon-exclamation-sign"
    />
     some-title
  </Component>
  <span>
    some-body
  </span>
  <div
    className="confirmation-check"
  >
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    <span>
       some-message
    </span>
  </div>
  <Component>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="danger"
      className="confirm-button"
      disabled={true}
      onClick={[Function]}
    >
      some-confirm
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      disabled={false}
      onClick={[MockFunction]}
    >
      Cancel
    </Button>
  </Component>
</ForemanModal>
`;
